pkg/sql/logictest/testdata/logic_test/pg_catalog

```diff
@@ -5338,3 +5338,44 @@ WHERE usename = 'regression_70180'
 ----
 usename           valuntil
 regression_70180  NULL
+
+# Test that pg_shadow doesn't include roles that can't login
+query B colnames
+SELECT (count(1) = 0) AS PASSED
+FROM pg_shadow
+WHERE EXISTS (
+  SELECT 1 FROM pg_authid WHERE oid = usesysid AND NOT rolcanlogin
+)
+----
+passed
+true
+
+# Test pg_shadow contents
+query TOBBBBTTT colnames
+SELECT * FROM pg_shadow ORDER BY usename;
+----
+usename                       usesysid    usecreatedb  usesuper  userepl  usebypassrls  passwd    valuntil                       useconfig
+admin                         2310524507  false        false     false    false         ********  NULL                           NULL
```

Comment:
shouldn't admin have `usesuper` privs? i thought admin was an admin role so has admin privileges...
@rafiss do you know?

Reply:
I think admin role is not giving true at UserHasAdminRole(ctx, username), and I think that is the case that useSuper have isRoot | isSuper

BUT, I think I missed to make isRoot for admin role.

Reply:
hm yeah, i think `admin` being a superuser should be fixed separately from this PR. it doesn't matter too much, since you can never login with `admin`, but we can fix it. https://github.com/cockroachdb/cockroach/issues/70779